shared/fs/banner/fileui-banner/container.js

```diff
@@ -0,0 +1,53 @@
+// @flow
+import Banner from './index'
+import * as FsGen from '../../../actions/fs-gen'
+import * as Types from '../../../constants/types/fs'
+import {connect, compose, lifecycle, setDisplayName, type TypedState} from '../../../util/container'
+import * as StateMappers from '../../utils/state-mappers'
+import {isMobile} from '../../../constants/platform'
+
+type OwnProps = {
+  path?: Types.Path,
+}
+
+const mapStateToProps = (state: TypedState) => {
+  const kbfsEnabled = StateMappers.mapStateToKBFSEnabled(state)
+  return {
+    kbfsEnabled,
+    showBanner: state.fs.flags.showBanner,
+    inProgress: state.fs.flags.fuseInstalling || state.fs.flags.kbfsInstalling || state.fs.flags.kbfsOpening,
+    showSecurityPrefs: !kbfsEnabled && state.fs.flags.kextPermissionError,
+  }
+}
+
+const mapDispatchToProps = (dispatch: Dispatch) => {
+  const uninstall = () => dispatch(FsGen.createUninstallKBFS())
+  return {
+    getFuseStatus: () => dispatch(FsGen.createFuseStatus()),
+    onDismiss: () => dispatch(FsGen.createSetFlags({showBanner: false})),
+    onInstall: () => dispatch(FsGen.createInstallFuse()),
+    onUninstall: () => dispatch(FsGen.createUninstallKBFSConfirm({onSuccess: uninstall})),
+    _openInFileUI: (path: Types.Path) => dispatch(FsGen.createOpenInFileUI({path: Types.pathToString(path)})),
+  }
+}
+
+const mergeProps = (stateProps, dispatchProps, {path}: OwnProps) => ({
+  ...stateProps,
+  ...dispatchProps,
```

Comment:
i think its better to just extract the props you're using instead of spreading. you're going to get _openInFileUI in your component if you do it like this